internal/setup/fetch.go

```diff
@@ -137,7 +137,7 @@ func extractTar(dataReader io.Reader, targetDir string) error {
 
 		//debugf("Extracting header: %#v", tarHeader)
 
-		err = fsutil.Create(&fsutil.CreateOptions{
+		err = fsutil.NewCreator().Create(&fsutil.CreateOptions{
```

Comment:
This is in a loop, creating and discarding the creator and its map on every iteration, not even keeping track of it. Is the idea here that the follow up PR will unify all of these calls and track these properly?

Reply:
The idea is that you have one creator per slice to know what files were created in each one. The other option will be cleaning the `creator.Created` map each iteration to know which files come from the new slice, I think that is more convoluted to follow. From the report PR:
```go
<loop over slice>
                creator := fsutil.NewCreator()
		err := deb.Extract(reader, &deb.ExtractOptions{
			Package:   slice.Package,
			Extract:   extract[slice.Package],
			TargetDir: targetDir,
			Globbed:   globbedPaths,
			Creator:   creator,
		})
		for _, entry := range creator.Created {
			report.AddEntry(slice, entry)
		}
<end of loop>
```

Reply:
That seems fine. What's a bit awkward is the case above where we have a tight loop around individual file extraction using a new creator every time.

Reply:
I will change that to the outer possible scope, you are right.